tests/unit/test_io_convert.py

```diff
@@ -65,62 +104,90 @@ def tearDown(self):
-        self.assertIsNotNone(str(exported_container))
-        # make sure we get a completely new object
-        self.assertNotEqual(id(self.container), id(exported_container))
-        # the name of the root container of a file is always 'root' (see h5tools.py ROOT_NAME)
-        # thus, ignore the name of the container when comparing original container vs read container
-        self.assertContainerEqual(self.container, exported_container,
-                                  ignore_name=self.IGNORE_NAME,
-                                  ignore_hdmf_attrs=self.IGNORE_HDMF_ATTRS,
-                                  ignore_string_to_byte=self.IGNORE_STRING_TO_BYTE)
-        # TODO May need to add further asserts here
-
-
-############################################
-# HDMF Common test harness
-###########################################
+        # determine and save the write and export paths
+        for write_path in self.WRITE_PATHS:
+            for export_path in self.EXPORT_PATHS:
+                container = self.setUpContainer()
+                container_type = container.__class__.__name__
+                if write_path is None:
+                    write_path = 'test_%s.hdmf' % container_type
+                if export_path is None:
+                    export_path = 'test_export_%s.hdmf' % container_type
+                self.filenames.append(write_path if isinstance(write_path, str) else write_path.path)
+                self.filenames.append(export_path if isinstance(export_path, str) else export_path.path)
+                # roundtrip the container
+                exported_container = self.roundtripExportContainer(
+                    container=container,
+                    write_path=write_path,
+                    export_path=export_path)
+                # assert that the rountrip worked correctly
+                message = "Using: write_path=%s, export_path=%s" % (str(write_path), str(export_path))
+                self.assertIsNotNone(str(container), message)  # added as a test to make sure printing works
+                self.assertIsNotNone(str(exported_container), message)
+                # make sure we get a completely new object
+                self.assertNotEqual(id(container), id(exported_container), message)
+                # the name of the root container of a file is always 'root' (see h5tools.py ROOT_NAME)
+                # thus, ignore the name of the container when comparing original container vs read container
+                self.assertContainerEqual(container,
+                                          exported_container,
+                                          ignore_name=self.IGNORE_NAME,
+                                          ignore_hdmf_attrs=self.IGNORE_HDMF_ATTRS,
+                                          ignore_string_to_byte=self.IGNORE_STRING_TO_BYTE)
+                # TODO: Add message to the assertContainerEqual call once the feature is released in HDMF
```

Comment:
this is about to be released in HDMF 3.5.0

Reply:
HDMF 3.5.0 was released so you can update the hdmf requirement and do this TODO in this PR or a new one

Reply:
Good catch. I updated the requirements and added the message here.